integration_tests/models/test_dates.yml

```diff
@@ -37,6 +37,12 @@ models:
             expression: "unix_epoch = {{ dbt_date.to_unixtimestamp('time_stamp_utc') }}"
         - dbt_utils.expression_is_true:
             expression: "time_stamp = {{ dbt_date.convert_timezone('time_stamp_utc') }}"
+        - dbt_utils.expression_is_true:
+            expression: "time_stamp = {{ dbt_date.convert_timezone('time_stamp_utc', source_tz='UTC') }}"
+        # - dbt_utils.expression_is_true:
```

Comment:
These tests don't work on platforms that don't support specifying a source timezone, so they'll remain commented out but stay in the code so it's easier to run them as needed.